### PR TITLE
Add "Built with XMTP" page

### PR DESCRIPTION
**Pull request by jhaaaa**
Just providing the foundation for the amazing "Built with XMTP" experience Rich has planned!

- Created the page in the Community section, per our chat: https://junk-range-possible-git-add-built-w-xmtp-xmtp-labs.vercel.app/built-with-xmtp
  - What do you think of the sidebar/left nav order? We can change it to suit your vision!
  - What do you think of the top nav drop-down menu - sparkle icon and the "Explore a showcase of apps built with XMTP" text? We can change that too...
  - Currently set up with a URL of `xmtp.org/built-with-xmtp`, as discussed. But we can easily change it...
- Created it as an MDX file. More info on how to use MDX and React in Docusaurus: https://docusaurus.io/docs/markdown-features/react
   - Also, I _think_ there may be more possibilities than MDX? For example, see Docusaurus' Showcase page built with Docusaurus: https://docusaurus.io/showcase. Here is the code: https://github.com/facebook/docusaurus/tree/main/website/src/pages/showcase. Not saying that it needs to look like this - but just some prior art for inspo!

# CREATE URL REDIRECTS

This PR changes a URL. After merging this PR, create the following URL redirects:

- Source: https://xmtp.org/community/community -> Target: https://xmtp.org/community
- Source: https://xmtp.org/community/community/ -> Target: https://xmtp.org/community

### CODE REVIEW

File: src/theme/Footer/index.js [hunk 1]
```diff
@@ -44,6 +44,7 @@ export default function FooterWrapper(props) {
           </div>
           <div className="flex w-[139px] justify-between">
             <ALink to="https://github.com/xmtp">
+              foo
```
Comment (jhaaaa) [inline src/theme/Footer/index.js:47]: Just checking - what are these two `foo`s - here and on line 60 - used for? =)

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (jhaaaa) [inline .vscode/launch.json:1]: Do we mean to merge these VS Code configurations to main? =)

File: src/components/BuiltWithXmtp/ListOfDevs.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      "image":"/img/builtWithXmtp/relay.png",
+      "description":"Relay’s goal is to provide open-source tools and products on top of XMTP, and create the best place for web3 identities.",
+      "github": "https://github.com/relaycc",
+      "twitter": "https://twitter.com/relay_eth"
+   },
+   {
+      "name":"Lensport",
+      "href":"https://lensport.io/",
+      "image":"/img/builtWithXmtp/lensport.png",
+      "description":"Discover, collect, and sell Lens Posts to invest in creators you support. Built on Lens Protocol.",
```
Comment (jhaaaa) [inline src/components/BuiltWithXmtp/ListOfDevs.json:48]: Do we want to add a GitHub URL here - or are we intentionally leaving it out for now? =)

File: src/components/BuiltWithXmtp/ListOfDevs.json [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      "image":"/img/builtWithXmtp/gamejutsu.png",
+      "description":"A gaming app that uses end-to-end encrypted messages as signed game moves between players.",
+      "github": "https://github.com/chainhackers/GameJutsu",
+      "twitter": "https://twitter.com/ChainHackerClan"
+   },
+   {
+      "name":"Buttrfly",
+      "href":"https://www.buttrfly.app/",
+      "image":"/img/builtWithXmtp/buttrfly-icon-rounded.png",
+      "description":"Decentralized social media app built for a mobile first experience on iOS. Built on Lens Protocol.",
```
Comment (jhaaaa) [inline src/components/BuiltWithXmtp/ListOfDevs.json:63]: Do we need to add a GitHub URL here - or are we intentionally leaving it out for now? =)

File: src/components/BuiltWithXmtp/index.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      {ListOfDevelopers.map(developer => 
+        <li className="relative group" key={developer.name}>
+          <a className="rounded border-solid rounded-md border-slate-200 dark:border-slate-600 flex justify-center py-6 mb-3" href={developer.href}>
+            <img className="max-h-20 max-w-20 min-h-20" src={useBaseUrl(developer.image)} alt={"this is an image of the" + developer.name + " icon"}></img>
+          </a>
+          <h4 className="my-1 text-xl mt-0">{developer.name}</h4>
+          <p className="text-base text-neutral-800 dark:text-neutral-300 line-clamp-3 mt-0 h-20">{developer.description}</p>
+          <div className='flex items-center'>
+              {developer.github && (
+                <a className="mb-2 mt-1 text-base font-semibold text-red-500 flex align-center mr-2"  href={developer.github}>Github<img src={useBaseUrl('/img/icons/github-icon-black.svg')} className="ml-2"/></a>
```
Comment (jhaaaa) [inline src/components/BuiltWithXmtp/index.js:21]: Tech writer tweak LOL - GitHub is camel-cased - I think this change will make it show up as "GitHub" on the Built with XMTP page, correct? =)

```suggestion
                <a className="mb-2 mt-1 text-base font-semibold text-red-500 flex align-center mr-2"  href={developer.github}>GitHub<img src={useBaseUrl('/img/icons/github-icon-black.svg')} className="ml-2"/></a>
```

**Review by jhaaaa — COMMENTED** [review]
I can't tell you how much I appreciate this page and all of the work and passion that went into it! 

Just a few questions and suggestions - with tech writer 💓! 😂🙏

Also I see it mentioned in the PR comments, so I think it is covered - but just in case, I wanted to mention dark mode. Do we plan to provide CSS for dark mode before sharing? TY!

File: src/components/BuiltWithXmtp/ListOfDevs.json (hunk 3 above)
Comment (richbustos) [reply src/components/BuiltWithXmtp/ListOfDevs.json:48]: leaving it out some projects dont have a public repo available

File: src/components/BuiltWithXmtp/ListOfDevs.json (hunk 4 above)
Comment (richbustos) [reply src/components/BuiltWithXmtp/ListOfDevs.json:63]: Same comment as above =)

File: src/components/BuiltWithXmtp/index.js (hunk 5 above)
Comment (richbustos) [reply src/components/BuiltWithXmtp/index.js:21]: perfect catch! Ill change it on my end and do a commit!

**Review by darickdang — APPROVED** [review]
lgtm

**Pull request merged**